backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Service/GasStationService.java

```diff
@@ -102,6 +102,23 @@ private Availability getAvailabilityByType(Set<Availability> availabilities, Int
         return null;
     }
 
+    public void updateAvailability(Integer gasStationId, Integer idFuelType, Integer idStatus) {
+        GasStation gasStation = gasStationRepository.findById(gasStationId)
+                .orElseThrow(() -> new RuntimeException("GasStation not found"));
+
+        if (idFuelType == null || idFuelType <= 0 || idStatus == null || idStatus <= 0) {
```

Comment:
Instead of this validation, use a repository validation for each one of the fields, same as the gasStation validation